internal/datanode/compaction_executor.go

```diff
@@ -87,13 +103,12 @@ func (c *compactionExecutor) executeTask(task compactor) {
 		)
 	}
 
-	c.executing.Delete(task.getPlanID())
 	log.Info("end to execute compaction", zap.Int64("planID", task.getPlanID()))
 }
 
 func (c *compactionExecutor) stopTask(planID UniqueID) {
-	task, loaded := c.executing.LoadAndDelete(planID)
-	if loaded {
+	task, ok := c.executing.Load(planID)
+	if ok {
 		log.Warn("compaction executor stop task", zap.Int64("planID", planID))
 		task.(compactor).stop()
 	}
```

Comment:
If `stop` invoked multiple times simultaneously , `waitgroup` might enter state of negative counter panic.

Reply:
Yes, this should stay LoadAndDelete